corehq/apps/ota/views.py

```diff
@@ -263,6 +259,10 @@ def get_restore_response(domain, couch_user, app_id=None, since=None, version='1
     )
 
     app = get_app_cached(domain, app_id) if app_id else None
+
+    error_msg, status_code = check_app_access(domain, couch_user, app)
```

Comment:
what if app is None?

Reply:
Hmm..I have not considered the case, but I guess we should not update the details then? 🤔

Reply:
Well currently the method `check_app_access` will raise an exception if app is None.
Checking with mobile team about when app_id would be missing for some other task, pining you there and then you can think about the necessary changes here.